fix: OPTIC-2157: Actions Menu Disappears at High Browser Zoom Levels

yyassi-heartex opened the pull request:
This pull request introduces changes to improve the alignment logic for dropdown elements in scenarios with limited viewport height. It adds a new parameter, `openUpwardForShortViewport`, to control whether dropdowns should open upward when there is insufficient space below. Debugging `console.log` statements have also been added for troubleshooting.

<img width="363" alt="image" src="https://github.com/user-attachments/assets/0b58bdb5-74c5-4b90-a766-3966ea709e99" />


### Alignment logic enhancements:
* [`web/libs/core/src/lib/utils/dom.ts`](diffhunk://#diff-5bcd905e2119e59538abf86ccee7613a1c338b995755a2dd89dd950164742bb6R99): Added a new parameter `openUpwardForShortViewport` to the `alignElements` function, defaulting to `true`. Updated the alignment logic to consider this parameter when determining whether to align dropdowns upward or downward. [[1]](diffhunk://#diff-5bcd905e2119e59538abf86ccee7613a1c338b995755a2dd89dd950164742bb6R99) [[2]](diffhunk://#diff-5bcd905e2119e59538abf86ccee7613a1c338b995755a2dd89dd950164742bb6L142-R145)

### Debugging improvements:
* [`web/libs/core/src/lib/utils/dom.ts`](diffhunk://#diff-5bcd905e2119e59538abf86ccee7613a1c338b995755a2dd89dd950164742bb6R163): Added a `console.log` statement to log alignment calculations, including `offsetTop`, `openUpwardForShortViewport`, and other related variables.
* [`web/libs/datamanager/src/components/Common/Dropdown/DropdownComponent.jsx`](diffhunk://#diff-70cc7355435ba12d350d0a0c3c9575cdce72a1dd4a37726b07d54248b7a06e74R27-R40): Added a `console.log` statement to log the value of `openUpwardForShortViewport` during position calculation in the `Dropdown` component.

yyassi-heartex committed:
fix: OPTIC-2157: Actions Menu Disappears at High Browser Zoom Levels

bmartel commented on web/libs/core/src/lib/utils/dom.ts:
```diff
@@ -159,6 +160,7 @@ export const alignElements = (
     offsetLeft = pos.horizontalRight;
     resultAlign[1] = "right";
   }
+  console.log("offsetTop", offsetTop, openUpwardForShortViewport, elem, target, resultAlign);
```
```suggestion
```

bmartel commented on web/libs/datamanager/src/components/Common/Dropdown/DropdownComponent.jsx:
```diff
@@ -24,18 +24,20 @@ export const Dropdown = React.forwardRef(({ animated = true, visible = false, ..
   const [visibility, setVisibility] = React.useState(visible ? "visible" : null);
 
   const calculatePosition = React.useCallback(() => {
+    console.log("calculatePosition", openUpwardForShortViewport);
```
```suggestion
```

yyassi-heartex committed:
should be keying off scrollY for height

yyassi-heartex committed:
minor fix for scrollX being used on height stuff

bmartel committed:
Update web/libs/datamanager/src/components/Common/Dropdown/DropdownComponent.jsx

bmartel committed:
use a prop from the component instead of hardcoding constrainHeight

bmartel committed:
make the diff easier

bmartel committed:
ensure constrainHeight is a dep

nass600 commented on web/libs/core/src/lib/utils/dom.ts:
```diff
@@ -96,6 +96,7 @@ export const alignElements = (
   align: Align = "bottom-left",
   padding = 0,
   constrainHeight = false,
+  openUpwardForShortViewport = true,
```
It feels it should not be an option or a variable but the normal behaviour meaning that, no matter the position the client selects, if there is overflow because of the viewport in any of the 4 directions, it should be added on a different direction by default

yyassi-heartex replied:
ahh i thought i responded to this - so I'm just adding this back in temporarily with the hope that when popover gets in dropdown starts leveraging that and we get rid of dom.ts altogether.
openUpwardForShortViewport is being used in some sports already - it was ignored during unification at some point so im just adding it back in for now

yyassi-heartex replied:
went ahead with merge since this fix is high priority

vladimirheartex commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14877667641)

vladimirheartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14877667641

vladimirheartex committed:
Merge branch 'develop' into 'fb-optic-2157/dropdown'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14877667641

yyassi-heartex commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14885827219)
> Successfully merged:  3 files changed, 20 insertions(+), 5 deletions(-)

yyassi-heartex committed:
Merge branch 'develop' into 'fb-optic-2157/dropdown'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14885827219
